src/lib/redux-cozy-client/slices/sharings.js

```diff
@@ -24,11 +29,18 @@ const documents = (state = [], action) => {
     case REVOKE_SHARING:
       const idx = state.findIndex(s => s.attributes.sharing_id === action.sharingId)
       if (idx === -1) return state
-      return [
-        ...state.slice(0, idx),
-        {...state[idx], attributes: {...state[idx].attributes, revoked: true}},
-        ...state.slice(idx + 1)
-      ]
+      const sharing = state[idx]
+      const loneRecipient = sharing.attributes.recipients.length === 1
```

Comment:
In this `REVOKE_SHARING` reducer, we check if there is only one recipient or a list of recipients...